Amplify/Core/Support/AmplifyInProcessReportingOperation.swift

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// An AmplifyOperation that emits InProcess values intermittently during the operation.
+///
+/// Unlike a regular `AmplifyOperation`, which emits a single Result at the completion of the operation's work, an
+/// `AmplifyInProcessReportingOperation` may emit intermediate values while its work is ongoing. These values could be
+/// incidental to the operation (such as a `Storage.downloadFile` operation reporting Progress values periodically as
+/// the download proceeds), or they could be the primary delivery mechanism for an operation (such as a
+/// `GraphQLSubscriptionOperation`'s emitting new subscription values).
+open class AmplifyInProcessReportingOperation<
```

Comment:
`AmplifyInProcessOperation`? would reduce size of name and still represent the same thing

Reply:
The part of me that values terseness approves of that idea, but "InProcessOperation" isn't the same as "InProcessReportingOperation". The first describes an operation that is in the middle of performing work, which actually applies to all of our operations. The second describes an operation that *tells you* when it is in the middle of performing work, which is the critical feature of this type.